cursorless-talon/src/private_api/extract_decorated_marks.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any
+
+from ..actions.bring_move import BringMoveTargets
+from ..actions.swap import SwapTargets
+from ..targets.target_types import (
+    ImplicitDestination,
+    ImplicitTarget,
+    ListDestination,
+    ListTarget,
+    PrimitiveDestination,
+    PrimitiveTarget,
+    RangeTarget,
+)
+
+
+def extract_decorated_marks(capture: Any) -> list[Any]:
+    match capture:
+        case PrimitiveTarget(mark=mark):
```

Comment:
This syntax is interesting.

Reply:
Yeah kinda nice I think

Reply:
I'm not sure how to read this. I totally follow the examples `case MyClass(value=5)` checks if the instance is of the correct type and the value field is the correct value. `mark=mark` I'm unsure how to interpret.

Reply:
It creates a local variable `mark` bound to whatever `mark` was in that dataclass. It's analogous to `const {x: x} = foo", where it constructs a local variable `x` bound to whatever was in the `x` field of `foo`

Reply:
So that syntax can be made for both comparisons and assignments. hmm. Sometimes python is a little bit to clever for my taste.

Reply:
Ha yeah I hadn't thought about that you can use it with constant. Yes maybe a bit clever 😅